crates/oxc_parser/src/modifiers.rs

```diff
@@ -178,20 +252,52 @@ impl ModifierKind {
         }
     }
 }
+impl From<Kind> for ModifierKind {
+    // NOTE (@DonIsaac) should we be using TryFrom here? or does unreachable!()
+    // give us compiler optimizations that we want?
+    fn from(kind: Kind) -> Self {
+        match kind {
+            Kind::Abstract => Self::Abstract,
+            Kind::Declare => Self::Declare,
+            Kind::Private => Self::Private,
+            Kind::Protected => Self::Protected,
+            Kind::Public => Self::Public,
+            Kind::Static => Self::Static,
+            Kind::Readonly => Self::Readonly,
+            Kind::Override => Self::Override,
+            Kind::Async => Self::Async,
+            Kind::Const => Self::Const,
+            Kind::In => Self::In,
+            Kind::Out => Self::Out,
+            Kind::Export => Self::Export,
+            Kind::Default => Self::Default,
+            Kind::Accessor => Self::Accessor,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl std::fmt::Display for ModifierKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
 
 impl<'a> ParserImpl<'a> {
-    pub(crate) fn eat_modifiers_before_declaration(&mut self) -> (ModifierFlags, Modifiers<'a>) {
+    pub(crate) fn eat_modifiers_before_declaration(&mut self) -> Modifiers<'a> {
```

Comment:
I'd eventually align all the method names with tsc's parser.ts as well so we know where to look when we try to debug parsing.

Reply:
I couldn't figure out what exactly was going on with this method so I left it as is, simply changing the return signature...

Reply:
> I couldn't figure out what exactly was going on with this method

Me neither, I didn't write it. We can get rid of it and port from tsc instead.

Reply:
Ive spent two hours staring at this code, `parse_modifers` and typescript's implementation. When I try to use `parse_modifiers` where this function is consumed, I get an infinite loop over `Kind::Const`. I can't figure it out. Any insights?

Reply:
Let's track this in the issue.